File: camerax_platform_interface/lib/src/core/plane_proxy.dart
import 'dart:typed_data';

final class PlaneProxy {
  final Uint8List value;
  final int pixelStride;
  final int rowStride;

  PlaneProxy({
    required this.value,
    required this.pixelStride,
    required this.rowStride,
  });
}

File: camerax_ios/ios/Classes/PlaneProxy.swift
//
//  PlaneProxy.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2024/8/22.
//

import Foundation

@objc public class PlaneProxy: NSObject {
    @objc public let rowStride: Int
    @objc public let pixelStride: Int
    @objc public let value: Data
    
    init(rowStride: Int, pixelStride: Int, value: Data) {
        self.rowStride = rowStride
        self.pixelStride = pixelStride
        self.value = value
    }
}
